.github/workflow/main.yml

```diff
@@ -0,0 +1,29 @@
+name: Release on NPM
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version-file: '.nvmrc'
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Build package
+        run: npm run build
+
+      - name: Update version and release it to NPM
+        env:
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+        run: touch .npmrc && echo "$NPM_TOKEN" >> .npmrc && npm realease && git push --follow-tags origin main && npm publish
```

Comment:
I found something in the npm docs. 
https://docs.npmjs.com/using-private-packages-in-a-ci-cd-workflow
So we could add the .npmrc to the repo. Also when we create it on the fly, we have to add a little bit more to the file.